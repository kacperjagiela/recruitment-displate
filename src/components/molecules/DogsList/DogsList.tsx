import { Stack } from '@chakra-ui/layout';

import { Button } from '~/components/atoms';
import Dog from '~/types/Dog';

interface Props {
    dogs: Dog[];
    onModalOpen: () => void;
    onDogButtonClick: (dog: Dog) => void;
}

const DogsList: React.FC<Props> = ({ dogs, onModalOpen, onDogButtonClick }: Props) => (
    <Stack direction={['column']}>
        {dogs.map((dog) => (
            <Button
                key={dog.name}
                buttonText={dog.name}
                onClick={() => {
                    onDogButtonClick(dog);
                    onModalOpen();
                }}
            />
        ))}
    </Stack>
);

export default DogsList;
