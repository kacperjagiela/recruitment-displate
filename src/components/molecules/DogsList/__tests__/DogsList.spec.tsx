import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import DogsList from '../DogsList';

describe('<DogsList />', () => {
    it('renders correctly', () => {
        const { container } = render(<DogsList dogs={[]} onModalOpen={jest.fn} onDogButtonClick={jest.fn} />);

        expect(container).toMatchSnapshot();
    });

    it('opens modal and handles dog button click on button click', () => {
        const onModalOpen = jest.fn();
        const onDogButtonClick = jest.fn();

        const { getByText } = render(
            <DogsList
                dogs={[{ name: 'testDog', url: '' }]}
                onModalOpen={onModalOpen}
                onDogButtonClick={onDogButtonClick}
            />,
        );

        fireEvent.click(getByText('testDog'));

        expect(onModalOpen).toHaveBeenCalledTimes(1);
        expect(onDogButtonClick).toHaveBeenCalledTimes(1);
    });
});
