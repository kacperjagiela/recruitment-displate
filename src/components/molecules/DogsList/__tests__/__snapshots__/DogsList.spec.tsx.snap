// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogsList /> renders correctly 1`] = `
<div>
  <div
    class="chakra-stack css-1tobaed"
  />
</div>
`;
